src/solvers/elasticity_solver.cpp

```diff
@@ -79,68 +68,64 @@ void ElasticitySolver::completeSetup()
 
   // Define the parallel linear form
 
-  l_form_ = new mfem::ParLinearForm(displacement_->space.get());
+  l_form_ = std::make_unique<mfem::ParLinearForm>(displacement_->space.get());
 
   // Add the traction integrator
   if (nat_bdr_.size() > 0) {
     for (auto& nat_bc : nat_bdr_) {
       l_form_->AddBoundaryIntegrator(new mfem::VectorBoundaryLFIntegrator(*nat_bc->vec_coef), nat_bc->markers);
     }
     l_form_->Assemble();
-    rhs_ = l_form_->ParallelAssemble();
+    rhs_.reset(l_form_->ParallelAssemble());
   } else {
-    rhs_  = new mfem::HypreParVector(displacement_->space.get());
+    rhs_  = std::make_unique<mfem::HypreParVector>(displacement_->space.get());
     *rhs_ = 0.0;
   }
 
   // Assemble the stiffness matrix
-  K_mat_ = K_form_->ParallelAssemble();
+  K_mat_ = std::unique_ptr<mfem::HypreParMatrix>(K_form_->ParallelAssemble());
 
   // Eliminate the essential DOFs
   for (auto& bc : ess_bdr_) {
-    K_e_mat_ = K_mat_->EliminateRowsCols(bc->true_dofs);
+    K_e_mat_.reset(K_mat_->EliminateRowsCols(bc->true_dofs));
```

Comment:
This continues to strike me as a very odd thing.  Looping over an object and destroying/assigning to the same object while keeping the last one to get there.  @jamiebramwell  assures me its fine but she may be messing with me.

Reply:
if its not needed why is it a member variable.

Reply:
These are the eliminated entries of the stiffness matrix. Since we don't currently support changing Dirichlet boundary conditions, it's OK to throw them away. However, we will need to keep these around (like in the `QuasiStaticSolve` function) if we want to handle changing BCs.